sdk/attestation/azure-security-attestation/src/private/jsonhelpers.cpp

```diff
@@ -0,0 +1,299 @@
+      return field[fieldName].get<int>();
+    }
+    return Azure::Nullable<int>();
+  }
+
+  // Serialization helpers.
+
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      std::string const& fieldValue,
+      std::string const& fieldName)
+  {
+    object[fieldName] = fieldValue;
+  }
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::Nullable<std::string> const& fieldValue,
+      std::string const& fieldName)
+  {
+    if (fieldValue.HasValue())
+    {
+      SetField(object, fieldValue.Value(), fieldName);
+    }
+  }
+
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      std::vector<std::string> const& fieldValue,
+      std::string const& fieldName)
+  {
+    object[fieldName] = fieldValue;
+  }
+
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::Nullable<std::vector<std::string>> const& fieldValue,
+      std::string const& fieldName)
+  {
+    if (fieldValue.HasValue())
+    {
+      SetField(object, fieldValue.Value(), fieldName);
+    }
+  }
+
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      int fieldValue,
+      std::string const& fieldName)
+  {
+    object[fieldName] = fieldValue;
+  }
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::Nullable<int> const& fieldValue,
+      std::string const& fieldName)
+  {
+    if (fieldValue.HasValue())
+    {
+      SetField(object, fieldValue.Value(), fieldName);
+    }
+  }
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      std::vector<int> const& fieldValue,
+      std::string const& fieldName)
+  {
+    if (!fieldValue.empty())
+    {
+      object[fieldName] = fieldValue;
+    }
+  }
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::Nullable<std::vector<int>> const& fieldValue,
+      std::string const& fieldName)
+  {
+    if (fieldValue.HasValue())
+    {
+      SetField(object, fieldValue.Value(), fieldName);
+    }
+  }
+
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::Nullable<Azure::DateTime> const& fieldValue,
+      std::string const& fieldName)
+  {
+    if (fieldValue.HasValue())
+    {
+      SetField(object, fieldValue.Value(), fieldName);
+    }
+  }
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::DateTime const& fieldValue,
+      std::string const& fieldName)
+  {
+    object[fieldName] = Azure::Core::_internal::PosixTimeConverter::DateTimeToPosixTime(fieldValue);
+  }
+
+  void JsonHelpers::SetField(
+      Azure::Core::Json::_internal::json& object,
+      Azure::Core::Json::_internal::json& fieldValue,
+      std::string const& fieldName)
+  {
+    object[fieldName] = fieldValue;
+  }
+}}}} // namespace Azure::Security::Attestation::_detail
```

Comment:
new line. But, if you just use JsonOptional, I don't think you need this file at all.